js/index.js

```diff
@@ -92,14 +92,19 @@ function handleMoveToSquare(event) {
          if (detectCheck(kingPiece, board)) {
              console.log("The opposing King has been placed in check.");
              if (detectCheckmate(kingPiece, board)) {
-                 console.log("The opposing King is in checkmate, game over.")
-             } else {
-                 currentPlayer = (currentPlayer === 0) ? 1 : 0; // switch turns here if opposing King isn't in checkmate.
-             }
-         }
-     }
+                 console.log("The opposing King is in checkmate, game over.");
+            } else {
+                currentPlayer = (currentPlayer === 0) ? 1 : 0; // switch turns here if opposing King isn't in checkmate.
+            }
+        }
+    }
+    const markedSquares = document.getElementsByClassName("markedSquare");
+    for (let i = 0; i < markedSquares.length; i++) {
+        markedSquares[i].removeEventListener("click", handleMoveToSquare); //Remove event listener here after the check/checkmate checks.
+    }
```

Comment:
Removing the event listeners on lines `101 - 104` still allows pieces to move to the marked squares after they are no longer marked since the `handleMoveToSquare()` event listener only fires when a piece is moved.